### PR TITLE
v1.23 blog post: Introducing COSI: Object Storage in Kubernetes

### PR DESCRIPTION
**Pull request by wlan0**
Blog post on new feature: COSI (container-object-storage-interface) for v1.23 release.

### CODE REVIEW

File: content/en/blog/_posts/2021-11-23-object-storage-support.md
Comment (sftim) [inline content/en/blog/_posts/2021-11-23-object-storage-support.md:24]: This doesn't make sense to me. The term “DevOps” comes from dev and ops _co-operating_ (eg [10+ Deploys Per Day: Dev and Ops Cooperation at Flickr](https://www.youtube.com/watch?v=LdOe18KhtT4)), and _not_ from making the delineation rigid.

How about:
```suggestion
   * Self Service - Allows teams to manage their workload's storage themselves, without waiting on other teams to make or approve infrastructure changes.
```

File: content/en/blog/_posts/2021-11-23-object-storage-support.md
Comment (sftim) [inline content/en/blog/_posts/2021-11-23-object-storage-support.md:42]:
```suggestion
## CustomResourceDefinitions for COSI {#api}
```

File: content/en/blog/_posts/2021-11-23-object-storage-support.md
Comment (sftim) [inline content/en/blog/_posts/2021-11-23-object-storage-support.md:44]:
```suggestion
The COSI API is centered around buckets, since they are the unit of abstraction for
provisioning object storage. There are three Kubernetes APIs (implemented as
[CustomResourceDefinitions](/docs/concepts/extend-kubernetes/api-extension/custom-resources/))
aimed at managing these:
```

**Review by sftim — COMMENTED** [review]
Thanks @wlan9

I'm not sure if any of the new components / code are part of Kubernetes itself, or if this is new out-of-tree code that's being released at the same time.

I'd like to make that detail clear if we can.

/hold
There doesn't seem to be a merged / tracked KEP for this; https://github.com/kubernetes/enhancements/pull/2813 is Open.

File: content/en/blog/_posts/2021-11-23-object-storage-support.md
Comment (sftim) [inline content/en/blog/_posts/2021-11-23-object-storage-support.md:56]: > Object Storage is always authenticated

In what way? There are widely used object storage technologies that support anonymous, unauthenticated access. For example, Amazon S3.

Comment (sftim) [inline content/en/blog/_posts/2021-11-23-object-storage-support.md:56]: For the KEP hyperlink, please link to https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/1979-object-storage-support/README.md

**Comment by sftim** [comment]
I found the KEP at https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/1979-object-storage-support/README.md

We should also hold for assigning a publication date.
/hold

**Comment by xing-yang** [comment]
We should wait for this KEP to be merged: https://github.com/kubernetes/enhancements/pull/2813

File: content/en/blog/_posts/2021-11-23-object-storage-support.md
Comment (karenhchu) [inline content/en/blog/_posts/2021-11-23-object-storage-support.md:4]:
```suggestion
date: 2021-12-14
```

Comment (karenhchu) [reply content/en/blog/_posts/2021-11-23-object-storage-support.md:4]: Per the request of the K8s 1.23 Release Comms Team, please update the publishing date to 12/14. Thank you!

Comment (karenhchu) [inline content/en/blog/_posts/2021-11-23-object-storage-support.md:4]:
```suggestion
date: 2021-12-14
```

**Review by karenhchu — CHANGES_REQUESTED** [review]
Per the request of the K8s 1.23 Release Comms Team, please update the publishing date to 12/14. Thank you!

File: content/en/blog/_posts/2021-11-23-object-storage-support.md
Comment (sftim) [reply content/en/blog/_posts/2021-11-23-object-storage-support.md:4]: Remember also to change the path (`content/en/blog/_posts/2021-11-23-object-storage-support.md`).

**Comment by sftim** [comment]
Not sure if https://github.com/kubernetes/enhancements/pull/2813 will merge or not; if it does not, I imagine we could repurpose that content for a future release?

**Comment by sftim** [comment]
This didn't make it for the v1.23 post-release blog series.

@wlan0 we'd welcome a similar article in the future - I hope you can open a new PR when you're ready to.
/close

**Pull request closed without merging**
